test/e2e/suite/conformance/certificates/suite.go

```diff
@@ -312,6 +313,32 @@ func (s *Suite) Define() {
 				Fail("Both signed certificates not signed by same private key")
 			}
 		})
+
+		It("should issue a basic, defaulted certificate for a single commonName and distinct dnsName defined by an ingress with annotations", func() {
+			ingClient := f.KubeClientSet.ExtensionsV1beta1().Ingresses(f.Namespace.Name)
+
+			name := "testcert-ingress"
+			secretName := "testcert-ingress-tls"
+
+			By("Creating an Ingress with the issuer name annotation set")
+			ingress, err := ingClient.Create(e2eutil.NewIngress(name, secretName, map[string]string{
+				"cert-manager.io/issuer":       issuerRef.Name,
+				"cert-manager.io/issuer-kind":  issuerRef.Kind,
+				"cert-manager.io/issuer-group": issuerRef.Group,
+			}, s.newDomain()))
+			Expect(err).NotTo(HaveOccurred())
+
+			certName := ingress.Spec.TLS[0].SecretName
+
+			By("Waiting for the Certificate to exist...")
+			Expect(e2eutil.WaitForCertificateToExist(
+				f.CertManagerClientSet.CertmanagerV1alpha2().Certificates(f.Namespace.Name), certName, time.Minute,
+			)).NotTo(HaveOccurred())
```

Comment:
May not be one for this PR if you cannot be bothered, but I'd love us to start using the gomego helpers for waiting in future...

```
Eventually(e2eutil.GetCertificate(...)).Should(Exist())
```

or something similar 😄

Reply:
:+1: